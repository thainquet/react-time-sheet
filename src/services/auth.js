import axios from 'axios'

const register = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/register`, params);
}

const login = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/login`, params);
}

const changePassword = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/changePass`, params);
}

const forgotPassword = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/forgotPass`, params);
}

export default {
  login, changePassword, forgotPassword, register
}