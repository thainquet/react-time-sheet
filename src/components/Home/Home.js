import React, { useGlobal } from 'reactn';
import { isLogin } from 'helpers/auth'

const Home = (props) => {
  if (!isLogin()) props.history.push('/login')
  const [username] = useGlobal('username');

  const handleLogout = () => {
    localStorage.clear()
    props.history.push('/login')
  }

  return (
    <div>
      <br />
      <b>Login successfully!! Welcome back, {username}</b>
      <br />
      <br />
      <button onClick={() => props.history.push('/update')}>Change password</button>
      <button onClick={() => handleLogout()}>Log out</button>
    </div>
  )
}


export default Home;

