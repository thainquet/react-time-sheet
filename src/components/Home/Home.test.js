import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { authServices } from 'services';
import Home from './Home';

const props = {
  history: []
}

it('should render correctly', () => {
  const wrapper = renderer.create(<Home {...props}/>).toJSON();
  expect(wrapper).toMatchSnapshot();
});

it('has 2 buttons', () => {
  const wrapper = mount(<Home {...props}/>);
  expect(wrapper.find('button')).toHaveLength(2);
  wrapper.unmount();
});
