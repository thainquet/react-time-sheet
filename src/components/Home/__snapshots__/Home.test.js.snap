// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <br />
  <b>
    Login successfully!! Welcome back, 
  </b>
  <br />
  <br />
  <button
    onClick={[Function]}
  >
    Change password
  </button>
  <button
    onClick={[Function]}
  >
    Log out
  </button>
</div>
`;
